### PR TITLE
cache descriptor refactors

### PR DESCRIPTION
**Pull request by godrei**
- cache.go and cache_test.go renamed to cache_path.go and cache_path_test.go
- cache descriptor and related functions moved to cache_descriptor.go
- cleanDuplicatePaths unused function removed
- CompareFilePathMaps func updates

### CODE REVIEW

File: cache_path_test.go [hunk 1]
```diff
@@ -221,14 +223,14 @@ func Test_parseIgnoreList(t *testing.T) {
...
-	for _, pth := range pths {
+func createDirStruct(t *testing.T, contentByPth map[string]string) {
+	for pth, content := range contentByPth {
 		dir := filepath.Dir(pth)
 		if err := os.MkdirAll(dir, 0777); err != nil {
 			t.Fatalf("failed to create dir: %s", err)
 			return
 		}
-		if _, err := os.Create(pth); err != nil {
+		if err := fileutil.WriteStringToFile(pth, content); err != nil {
```
Comment (BirmacherAkos) [inline cache_path_test.go]: You changed the method, but the error message is the same.
Could you update the error message as well in your next PR?

File: main.go [hunk 2]
```diff
@@ -652,7 +567,7 @@ func main() {
 		}
 		cacheModel.FilePathMap = filePathsMap
 
-		recacheRequired, err = cacheModel.CompareFilePathMaps(filePathsMap)
+		recacheRequired = CompareFilePathMaps(cacheModel.PreviousFilePathMap, filePathsMap, cacheModel.DebugMode)
```
Comment (BirmacherAkos) [inline main.go]: The new `CompareFilePathMaps()` method will not return error, but you still check the `err != nil` 
In this case it's the previous error which was already checked at:
https://github.com/bitrise-steplib/steps-cache-push/blob/2a5944daa914ebf9fc6f83a7549d9b12b99a0e63/main.go#L564

File: cache_path_test.go (hunk 1 above)
Comment (godrei) [reply cache_path_test.go]: good catch, thx

File: main.go (hunk 2 above)
Comment (godrei) [reply main.go]: another good catch

**Comment by lpusok** [comment]
lgtm

**Pull request merged**